lib/app/router/providers/go_router_provider.r.dart

```diff
@@ -38,10 +39,16 @@ GoRouter goRouter(Ref ref) {
       final initState = ref.read(initAppProvider);
       final isSplashAnimationCompleted = ref.read(splashProvider);
       final forceUpdateRequired = ref.read(forceUpdateProvider).valueOrNull.falseOrValue;
+      final isAndroidSoftUpdateRequired = ref.read(androidSoftUpdateProvider).isUpdateAvailable;
```

Comment:
I think we need to listen for this provider if we want to get notified when it changes the value in `lib/app/router/app_router_listenable.dart`